pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -49,8 +52,44 @@ type RemoteError struct {
 }
 
 // New returns an authenticator which will make a basic auth call to the given url.
-func New(providerName string, url string, mapper authapi.UserIdentityMapper) authenticator.Password {
-	return &Authenticator{providerName, url, mapper}
+func New(providerName string, url string, client *http.Client, mapper authapi.UserIdentityMapper) authenticator.Password {
+	if client == nil {
+		client = http.DefaultClient
+	}
+	return &Authenticator{providerName, url, client, mapper}
+}
+
+// NewClient returns an http.Client for the given ca and client cert.
+// If no ca or client cert is specified, http.DefaultClient is returned.
+func NewClient(ca string, certFile string, keyFile string) (*http.Client, error) {
```

Comment:
Why separate these two methods?  I get the general utility of this function, but if that's the case, why not create something in a more general location?